### PR TITLE
Migrate Optimism `uniswap_v3.view_pools` to V2

### PR DESCRIPTION
**Pull request by chuxinh**
Migrating this abstraction from V1 to V2: https://github.com/duneanalytics/abstractions/tree/master/optimism2/uniswap_v3
cc: @MSilb7 

Messed up with my branches so create a new one from: https://github.com/duneanalytics/abstractions/pull/1380

*For Dune Engine V2*
I've checked that: 

* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory): https://dune.com/queries/1170735?d=1
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`

When you are ready for a review, tag duneanalytics/data-experience. We will re-open your forked pull request as an internal pull request. Then your spells will run in dbt and the logs will be avaiable in Github Actions DBT Slim CI. This job will only run the models and tests changed by your PR compared to the production project.

### CODE REVIEW

**Comment by jeff-dude** [comment]
thank you @chuxinh for the spells!

i'm planning to do a manual deployment (ideally today) with some recent new spells. once that is successful, i can return and help review your PRs (including #1390)

File: spellbook/models/uniswap/optimism/uniswap_optimism_schema.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  - name: uniswap_optimism_pools
+    meta:
+      blockchain: optimism
+      project: uniswap
+      contributors: MSilb7, chuxinh
+    config:
+      tags: ['optimism','uniswap','pools']
+    description: >
+        A canonical table containing all Uniswap pools created on Optimism.
+        tests:
```
Comment (dot2dotseurat) [inline spellbook/models/uniswap/optimism/uniswap_optimism_schema.yml:13]: @chuxinh it doesn't seem like this test is triggering anything. I ran the build command and no tests ran. I think you need to move it under a column. Is pool the column you'd like to check for uniqueness?

```
models:
  - name: uniswap_optimism_pools
    meta:
      blockchain: optimism
      project: uniswap
      contributors: MSilb7, chuxinh
    config:
      tags: ['optimism','uniswap','pools']
    description: >
        A canonical table containing all Uniswap pools created on Optimism.
    columns:
      - &pool
        name: pool
        description: "Uniswap liquidity pool contract address"
        tests:
          - unique

**Comment by dot2dotseurat** [comment]
LGTM after the test is moved under the pool column
```
19:38:34  Running with dbt=1.1.1
19:38:34  Found 86 models, 44 tests, 0 snapshots, 0 analyses, 460 macros, 0 operations, 19 seed files, 44 sources, 0 exposures, 0 metrics
19:38:34  
19:40:43  Concurrency: 1 threads (target='dev')
19:40:43  
19:40:43  1 of 2 START view model dbt_meghan_uniswap_v3_optimism.pools ................... [RUN]
19:40:46  1 of 2 OK created view model dbt_meghan_uniswap_v3_optimism.pools .............. [OK in 3.18s]
19:40:46  2 of 2 START test unique_uniswap_optimism_pools_pool ........................... [RUN]
19:40:51  2 of 2 PASS unique_uniswap_optimism_pools_pool ................................. [PASS in 4.92s]
19:40:52  
19:40:52  Finished running 1 view model, 1 test in 137.19s.
19:40:52  
19:40:52  Completed successfully
19:40:52  
19:40:52  Done. PASS=2 WARN=0 ERROR=0 SKIP=0 TOTAL=2

File: spellbook/models/uniswap/optimism/uniswap_optimism_schema.yml (hunk 1 above)
Comment (chuxinh) [reply spellbook/models/uniswap/optimism/uniswap_optimism_schema.yml]: Oh yea, thanks for this! 
Have it updated

**Review by dot2dotseurat — APPROVED** [review]
lgtm!

**Pull request merged**